generators/profilegen/src/profile_generator.rb

```diff
@@ -0,0 +1,225 @@
+    @output_dir = profile['output_dir']
+    @individual_gem_profile = profile['individual_gem_profile'].nil?? false: true
+    @dir_metadata = dir_metadata
+    @module_name = ''
+    @file_names, @model_types, @operation_types = [], [], []
+    @spec_includes, @class_names = [], []
+    @versions_clients_mapper = {}
+  end
+
+  def generate
+    generate_profile_sdk
+  end
+
+  private
+  def generate_profile_sdk
+    generate_modules
+    generate_client
+    generate_module_definition
+  end
+
+  def generate_modules
+    @resource_provider_types.each do |resource_provider, resource_types_obj|
+      @module_require = @dir_metadata[resource_provider]['module_require']
+      @module_name    = @dir_metadata[resource_provider]['module_name']
+      @spec_includes << @module_require
+      @class_name     = get_ruby_specific_resource_type_name(resource_provider)
+      @class_names   << @class_name
+
+      resource_types_obj.each do |resource_type_version, resource_types|
+        base_file_path = @dir_metadata[resource_provider]['path'] + get_sdk_path + "#{resource_type_version}/generated/#{@module_require}.rb"
+        require base_file_path
+
+        resource_types.each do |resource_type|
+          generate_operation_types(resource_provider, resource_type, resource_type_version)
+        end
+
+        generate_model_types(resource_provider, resource_type_version)
+      end
+
+      file = get_module_file resource_provider
+      file.write(get_renderer(ProfileTemplates.module_template))
+      @model_types, @operation_types, @versions_clients_mapper = [], [], {}
+      @management_client = ''
+    end
+  end
+
+  #
+  # Method to generate the client file
+  # Gets the client template and renders it
+  #
+  def generate_client
+    file = get_client_file
+    file.write(get_renderer(ProfileTemplates.client_template))
+  end
+
+  def generate_module_definition
+    file = get_module_definition_file
+    file.write(get_renderer(ProfileTemplates.module_definition_template))
+  end
+
+  #
+  # Method to get renderer
+  #
+  # @param template for which the renderer will be created
+  #
+  def get_renderer(template)
+    renderer = ERB.new(template, 0, '-%>')
+    renderer.result(get_binding)
+  end
+
+  def generate_operation_types(resource_provider, resource_type, resource_type_version)
+    namespace = @dir_metadata[resource_provider]['namespace'] + "::#{get_version(resource_type_version)}"
+    namespace = namespace.gsub(/\./, '_' )
+    obj = Module.const_get(namespace)
+    operations = []
+    obj.constants.select do |const_name|
+      if((obj.const_get(const_name).instance_of?Class))
+        if(const_name.to_s.end_with?('ManagementClient') || const_name.to_s == 'ManagementLinkClient' || const_name.to_s == 'ManagementLockClient' || const_name.to_s == 'FeatureClient' || const_name.to_s == 'PolicyClient' || const_name.to_s == 'SubscriptionClient')
+          @management_client = obj.name + '::' + const_name.to_s
```

Comment:
minor: we should leverage string interpolation in ruby instead of concatenation

Reply:
Removed it here and in other places.